homework/index.js

```diff
@@ -30,14 +32,145 @@
+          height: 80,
+        });
+        createAndAppend('div', tr, { text: name, class: 'col-30 description' });
+        const td3 = createAndAppend('div', tr, { class: 'col-30' });
+        createAndAppend('label', td3, { text: contributionsCount });
+      }
+    });
+  }
+  function bindRepositoryDetails(selectedRepository) {
+    const { description } = selectedRepository.description;
+    const { forks } = selectedRepository.forks;
+    const repositoryName = selectedRepository.name;
+    const updatedOn = new Date(selectedRepository.updated_at).toLocaleString();
+    const repoURL = selectedRepository.html_url;
+
+    const infoContainerDiv = document.getElementById('repo-info-container');
+    const repoInfodiv = createAndAppend('div', infoContainerDiv, {
+      class: 'repo-info',
+    });
+    const div = createAndAppend('div', repoInfodiv);
+    const table = createAndAppend('div', div, { class: 'grid' });
+
+    const tr1 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr1, { text: 'Repository:', class: 'col-50' });
+    const repoNameCol = createAndAppend('div', tr1, { class: 'col-50' });
+    createAndAppend('a', repoNameCol, {
+      href: repoURL,
+      target: '_blank',
+      text: repositoryName,
+    });
+
+    const tr2 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr2, { text: 'Description:', class: 'col-50' });
+    createAndAppend('div', tr2, { text: description, class: 'col-50' });
+
+    const tr3 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr3, { text: 'Forks:', class: 'col-50' });
+    createAndAppend('div', tr3, { text: forks, class: 'col-50' });
+
+    const tr4 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr4, { text: 'Updated:', class: 'col-50' });
+    createAndAppend('div', tr4, { text: updatedOn, class: 'col-50' });
+  }
+
+  function repositoryChanged(event) {
+    const infoElement = document.getElementById('repo-info-container');
+    infoElement.innerHTML = '';
+    const repositoryIndex = parseInt(event.target.selectedOptions[0].value, 10);
+    const selectedRepo = repositoryList[repositoryIndex];
+    bindRepositoryDetails(selectedRepo);
+    bindContributions(selectedRepo.contributors_url);
+  }
+
+  function createDropDown(parentElement, repositories) {
+    const topNavDiv = createAndAppend('div', parentElement, { class: 'topnav' });
+    const div = createAndAppend('div', topNavDiv, { class: 'header-container' });
+    const div1 = createAndAppend('div', div, { class: 'col-50' });
+    createAndAppend('a', div1, { text: 'HYF Repositories', href: '#' });
+    const repoDiv = createAndAppend('div', div, { class: 'col-50' });
+    const repoDropDown = createAndAppend('select', repoDiv, {
+      id: 'ddlRepository',
+      class: 'select-css',
+    });
+    createAndAppend('option', repoDropDown, { text: 'Select Repository' });
+    repositories.sort((a, b) => {
```

Comment:
I suggested the use of `localeCompare` in my previous feedback. The problem of your custom implementation is that it may not handle language specific special characters such as accented letter in French etc. And I know that for instance Arabic doesn't have uppercase of lowercase.

The `.localeCompare()` method use the language settings from the user's computer configuration. For instance, on a computer configured to use the French language and regional settings, `localeCompare` will sort according to the French dictionary order.

Reply:
done